Formula/s/semgrep.rb

```diff
@@ -29,7 +29,22 @@ class Semgrep < Formula
   depends_on "coreutils"=> :build
   depends_on "dune" => :build
   depends_on "ocaml" => :build
-  depends_on "opam" => :build
+
+  # Pin opam to 2.1.5 because >2.1.6 fails on linux, having problems
+  # with conf-libcurl.1 conf-libev.4-12, ....
+  #
+  # More details about the error here:
+  # https://github.com/Homebrew/homebrew-core/pull/176636#issuecomment-2212640312
+  #
+  # It doesn't fail on macosx though.
+  #
+  # While we still try to figure this out, pin opam to 2.1.5 for now,
+  # so the release goes through. Since opam is a package manager (not
+  # a library), and only needed during build time, not having the most
+  # recent version is a tolerable issue compared to not being able to
+  # release.
+  depends_on "opam" => ["2.1.5", :build]
```

Comment:
This syntax only works for XCode. This pin needs to be fixed upstream.

Reply:
Thanks for the pointer. Also thanks for the style fix, I was confused with how I quoted things.

I'm going to try a different approach, and will do a force push on this PR.